app/src/test/java/com/example/trainingintive/dog_images_feature/presentation/DogImageViewModelTest.kt

```diff
@@ -36,20 +37,17 @@ class DogImageViewModelTest : RulesForTests() {
 
     @Test
     fun `after successfully download, dogimage should be inserted into local database`() {
-        DogImageViewModel(repository, schedulersForTests)
+        DogImageViewModel(repository, schedulersForTests, mainNavigator)
 
         verify { repository.insert(sampleDog) }
     }
 
     @Test
-    fun `if repository returns error livedata should contain message`() {
-        val errorMessage = "Unable to resolve host \"dog.ceo\": No address associated with hostname"
-        every { repository.getDogImageUrl() } returns Single.error(Exception(errorMessage))
+    fun `if repository returns error vm should send error event`() {
+        every { repository.getDogImageUrl() } returns Single.error(Exception(""))
 
-        val tested = DogImageViewModel(repository, schedulersForTests)
+        DogImageViewModel(repository, schedulersForTests, mainNavigator)
 
-        tested.dog.observeForever { result ->
-            result shouldBeEqualTo errorMessage
-        }
+        verify { mainNavigator.sendEvent(ofType(MainScreenEvent.Error::class)) }
```

Comment:
Here you are only verifying that error of type MainScreenEvent.Error is send to navigator. Please add additional test to check if `toErrorTextId` is called on throwable returned from repo